sqlmesh/core/context.py

```diff
@@ -1625,6 +1630,7 @@ def create_external_models(self) -> None:
                 adapter=self._engine_adapter,
                 state_reader=self.state_reader,
                 dialect=config.model_defaults.dialect,
+                gateway=self.gateway,
```

Comment:
This is the only change I made to this file. The rest are due to `pre-commit` reformatting the file

Reply:
It's weird that the current build passes on main without these other changes.

Reply:
Could it be that pre-commit using a different version / config of `ruff`?

Reply:
Possibly. I'm not sure what version it's meant to be using, but here is the version it is using:

```
$ ruff --version
ruff 0.4.7
```

Reply:
I have the same one

Reply:
I think I understand what happened. My IDE originally had the `black` formatter enabled and I must have saved this file before I turned it off.

`ruff` apparently didnt mind the changes `black` made so it didn't change them back. 

I've rebased this file from main and the formatting changes have gone

Reply:
Thanks!